lib/internal/modules/esm/module_job.js

```diff
@@ -186,6 +195,20 @@ class ModuleJob {
     }
   }
 
+  runSync() {
+    assert(this.module instanceof ModuleWrap);
```

Comment:
Why do we need any assertion?

Reply:
It's a sincere question. Is this here because you were debugging and forgot to delete it, or are there cases where this throws? If the latter, what cases? Can you add a comment here mentioning what they are?

Reply:
> If the latter, what cases?

Like the error message says, it can be caused "by either a bug in Node.js or incorrect usage of Node.js internals." Obviously it should never happen, but that's the point of having an assertion. I don't think a comment is necessary.

Reply:
If there are no cases when it can happen, other than us introducing bugs into the code, then it should probably be a test rather than something that runs on runtime for every module. This is just an unnecessary line of code that slows things down.

Reply:
Couldn't that be said about any other assertion in the codebase?

Reply:
Yeah, probably. Presumably that’s why there are very few assertions in the codebase.